src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/mobile/MobileAdsBaseObject.java

```diff
@@ -102,6 +110,92 @@ public void verifyImgAdLoadedInSlot(String slotName, String expectedImg) {
     PageObjectLogging.log("AdInSlot", "Ad found in slot", true, driver);
   }
 
+  /**
+   * Scrolls to the main page link (to hide top bars) and clicks the link
+   */
+  public void mercuryNavigateToMainPage() {
+    if (checkIfElementOnPage(MERCURY_WIKI_TITLE_SELECTOR)) {
+      WebElement mainPageLink = driver.findElement(By.cssSelector(MERCURY_WIKI_TITLE_SELECTOR));
+      scrollToElement(mainPageLink);
+
+      PageObjectLogging.log(
+        "mercuryNavigateToMainPage()",
+        String.format(
+          "Clicking main page link: %s (%s)",
+          mainPageLink.getText(),
+          mainPageLink.getAttribute("href")
+        ),
+        true,
+        driver
+      );
+
+      mainPageLink.click();
+    }
+  }
+
+  /**
+   * Gets list of /wiki/* links on page and clicks the first valid
+   */
+  public void mercuryNavigateToAnArticle() {
+    if (checkIfElementOnPage(WIKI_LINK_SELECTOR)) {
+      String notArticlePattern = ".*\\/wiki\\/.*\\:.*";
+      List<WebElement> links = driver.findElements(By.cssSelector(WIKI_LINK_SELECTOR));
+
+      PageObjectLogging.log(
+        "mercuryNavigateToAnArticle()",
+        "Found " + links.size() + " internal links",
+        true,
+        driver
+      );
+
+      for(WebElement link : links) {
+        String href = link.getAttribute("href");
+        Boolean isArticle = !href.matches(notArticlePattern);
+        Boolean isMainPage = href.equals(getMercuryMainPageHref());
+
+        if (isArticle && !isMainPage) {
+          PageObjectLogging.log(
+            "mercuryNavigateToAnArticle()",
+            String.format(
+              "Link found, clicking: %s (%s)",
+              link.getText(),
+              link.getAttribute("href")
+            ),
+            true,
+            driver
+          );
+
+          scrollToElement(link);
+          link.click();
+          return;
+        }
+      }
+    } else {
+      PageObjectLogging.logWarning("mercuryNavigateToAnArticle()", "No links on main page!");
+    }
+  }
+
+  public void mercuryWaitForPageToLoad() {
+    driver.manage().timeouts().implicitlyWait(250, TimeUnit.MILLISECONDS);
+    try {
+      PageObjectLogging.log("mercuryWaitForPageToLoad", "Waiting till loaded...", true, driver);
+      wait.until(
+        ExpectedConditions.invisibilityOfElementLocated(By.cssSelector(MERCURY_LOADING_OVERLAY))
+      );
```

Comment:
It won't work since Mercury is a single page application. That means `DOMReady` event is being sent after first page load and with all other DOM is always ready :japanese_ogre: 

I tried using `AdsBaseObject::waitPageLoaded()` anyway and it passes sometimes and sometimes it fails with:
`org.openqa.selenium.WebDriverException: unknown error: Element is not clickable at point (285, 70). Other element would receive the click: <div id="ember429" class="ember-view loading-overlay">...</div>`

That's why I'm staying with the method. But I'm adding a description and changing its name.